bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,482 @@
+
+        if (!jsonResponse.contains("error") && jsonResponse.contains("total")) {
+            try {
+                JsonObject Object = (JsonObject) new JsonParser().parse(jsonResponse);
+                JsonObject myObject = Object.getAsJsonObject("data").getAsJsonObject("viewer").getAsJsonObject("home")
+                        .getAsJsonObject("currentSubscription").getAsJsonObject("priceInfo").getAsJsonObject("current");
+
+                updateState(CURRENT_TOTAL, new DecimalType(myObject.get("total").toString()));
+                updateState(CURRENT_STARTSAT, new StringType(myObject.get("startsAt").toString()));
+
+                if (getThing().getStatus() == ThingStatus.OFFLINE
+                        || getThing().getStatus() == ThingStatus.INITIALIZING) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Error communicating with Tibber API");
+            }
+        } else if (jsonResponse.contains("error")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error in response from Tibber API");
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unexpected response from Tibber");
+        }
+        return;
+    }
+
+    public void getURLDaily(String url) throws IOException {
+        InputStream inputStream = priceInfo.getDailyInputStream(configuration.getHomeid());
+        String jsonResponse = HttpUtil.executeUrl("POST", url, httpHeader, inputStream, null, REQUEST_TIMEOUT);
+        logger.debug("API response (Daily): {}", jsonResponse);
+
+        if (!jsonResponse.contains("error") && jsonResponse.contains("cost")) {
+            try {
+                JsonObject Object = (JsonObject) new JsonParser().parse(jsonResponse);
+                JsonObject myObject = (JsonObject) Object.getAsJsonObject("data").getAsJsonObject("viewer")
+                        .getAsJsonObject("home").getAsJsonObject("daily").getAsJsonArray("nodes").get(0);
+
+                updateState(DAILY_FROM, new StringType(myObject.get("from").toString()));
+                updateState(DAILY_TO, new StringType(myObject.get("to").toString()));
+
+                updateChannel(DAILY_COST, myObject.get("cost").toString());
+                updateChannel(DAILY_CONSUMPTION, myObject.get("consumption").toString());
+
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Error communicating with Tibber API");
+            }
+        } else if (jsonResponse.contains("error")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error in response from Tibber API");
```

Comment:
You should probably either log response error message or include it in the status.

Reply:
Corrected.